cmd/openshift-install/create.go

```diff
@@ -267,6 +274,10 @@ func runTargetCmd(targets ...asset.WritableAsset) func(cmd *cobra.Command, args
 
 		err := runner(rootOpts.dir)
 		if err != nil {
+			if cmd.Name() == "cluster" {
+				logrus.Error(err)
+				logrus.Exit(exitCodeInfrastructureFailed)
```

Comment:
This will categorize errors that are not infrastructure errors as such. One example is if the install-config.yaml is invalid.

Reply:
You could have the Cluster.Generate return a specific error type when the failure is from applying the terraform.